woocommerce/woocommerce-ios
# Shipping Labels: Display the last selected package in Package Details

Closes: #3940

## Description

This PR sets the shipping label package as the last selected package (from the API) by default. This displays the package name in the Package Selected field in the Package Details screen and sets it as the selected package (so it is already selected when you open the packages list).

## Changes

* Adds `syncShippingLabelAccountSettings` to the shipping label form screen, to sync the shipping label account settings. These settings include the last selected package ID, as well as other settings we'll use later in the shipping label flow (including payment methods).
* Moves `syncPackageDetails` to the shipping label form screen (from package details) so that package details are already synced when we open the Package Details screen, and passes the response to the Package Details VM.
* Adds `accountSettingsResultsController` to retrieve the shipping label account settings from storage on the Package Details screen.
* Sets the selected package ID to the last selected package ID with `setDefaultPackage`, so the last selected package is selected and displayed.
* Updates the Package Details screen to display `selectedPackageName` — previously it displayed `selectedPackageID`, which isn't the same as the name (title) for predefined packages.

<img src="https://user-images.githubusercontent.com/8658164/116578173-0cd4bb00-a909-11eb-8859-36ee0b305c70.gif" width="300px">


## Testing

Setup:

* If you haven't purchased a shipping label on your store before, the Package Selected field will be blank. You can test first with this scenario and then test after purchasing a shipping label.
* You can follow the internal testing guide at PCYsg-wev-p2 to purchase a shipping label before testing. Alternately, you can use a proxy app like Charles Proxy and edit the API response from the `/wc/v1/connect/account/settings` endpoint to fake a last selected package value; with this approach, enter one of your existing package IDs as the `last_box_id` in `userMeta` in that response.

To test:

1. Launch the app in debug mode (because of the feature flag).
2. Open an order detail.
3. Press the button "Create Shipping Label".
4. Press the button "Continue" in Create Shipping Label form, under the `Ship From` cell. Confirm the address.
5. Press the button "Continue" in Create Shipping Label form, under the `Ship To` cell. Confirm the address.
6. Press the button "Continue" in Create Shipping Label form, under the `Package Details` cell.
7. Confirm the `Package Selected` field on the Package Details screen displays your last selected package by default.
8. Tap on Package Selected and confirm the last selected package is selected in the package list.
9. Select a different package and tap Done. Confirm `Package Selected` now shows the new package selection.

## Submitter Checklist

Update release notes:
